### PR TITLE
Add decompositions for zero_, fill_, new_full, new_zeros, new_ones

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82332

This makes symbolic tracing tests for logsigmoid and xlogy start working again.

While I'm at it, add pin_memory and layout kwargs to empty; but they
don't actually do anything and raise an error if they are non standard.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by ezyang** [comment]
@jansel I cc'ed you because even though these decomps are done in the prevailing PrimTorch style it is not really clear if inductor would actually like this style haha

**Comment by ezyang** [comment]
cc @zou3519 because these decomps make the composite compliance problem worse

**Comment by zou3519** [comment]
> cc @zou3519 because these decomps make the composite compliance problem worse

We'll figure it out eventually, but no need to block this PR on figuring out how to make them composite compliant

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -2998,13 +3012,60 @@ def new_empty(
...
+def new_zeros(
+    a: TensorLikeType,
+    size: ShapeType,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    layout: Optional[torch.layout] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+) -> TensorLikeType:
+    r = a.new_empty(size, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory)
```
Comment (eellison) [inline torch/_refs/__init__.py:3025]: is it worth having `new_full` as a prim operator which would cover new_zeros/new_ones/new_full without requiring two operators & mutation

**Review by eellison — APPROVED** [review]
LGTM

File: torch/_refs/__init__.py (hunk 1 above)
Comment (ezyang) [reply torch/_refs/__init__.py]: Not sure but I can always route these to `new_full`, at least...

**Pull request closed without merging**